src/openMVG/tracks/tracks.hpp

```diff
@@ -130,68 +150,75 @@ struct TracksBuilder
   const MapNodeToIndex & getReverseMap() const {return _map_nodeToIndex;}
 
   /// Build tracks for a given series of pairWise matches
-  bool Build( const PairWiseMatches &  map_pair_wise_matches)
+  bool Build( const PairwiseMatches &  pairwiseMatches)
   {
-    typedef std::set<indexedFeaturePair> SetIndexedPair;
+    typedef std::set<IndexedFeaturePair> SetIndexedPair;
     // Set of all features of all images: (imageIndex, featureIndex)
     SetIndexedPair allFeatures;
     // For each couple of images
-    for (PairWiseMatches::const_iterator iter = map_pair_wise_matches.begin();
-      iter != map_pair_wise_matches.end();
-      ++iter)
+
+    // Make the union according the pair matches
+    for(const auto& matchesPerDescIt: pairwiseMatches)
     {
-      const size_t & I = iter->first.first;
-      const size_t & J = iter->first.second;
-      // Features correspondences between I and J image.
-      const std::vector<IndMatch> & vec_FilteredMatches = iter->second;
+      const size_t & I = matchesPerDescIt.first.first;
+      const size_t & J = matchesPerDescIt.first.second;
+      const MatchesPerDescType& matchesPerDesc = matchesPerDescIt.second;
 
-      // Retrieve all features
-      for( size_t k = 0; k < vec_FilteredMatches.size(); ++k)
+      for(const auto& matchesIt: matchesPerDesc)
       {
-        allFeatures.insert(std::make_pair(I,vec_FilteredMatches[k]._i));
-        allFeatures.insert(std::make_pair(J,vec_FilteredMatches[k]._j));
+        const features::EImageDescriberType descType = matchesIt.first;
+        const IndMatches& matches = matchesIt.second;
+        // We have correspondences between I and J image index.
+        for(const IndMatch& m: matches)
+        {
+          IndexedFeaturePair pairI(I, KeypointId(descType, m._i));
+          IndexedFeaturePair pairJ(J, KeypointId(descType, m._j));
+          allFeatures.insert(pairI);
+          allFeatures.insert(pairJ);
+        }
       }
     }
 
     // Build the node indirection for each referenced feature
     MapIndexToNode map_indexToNode;
+#ifdef HAVE_BOOST

```

Comment:
#if OPENMVG_IS_DEFINED(OPENMVG_HAVE_BOOST)  and #include <openMVG/config.hpp> at the very beginning